packages/lib-sourcify/README.md

```diff
@@ -58,132 +185,499 @@ import {
 } from "@ethereum-sourcify/lib-sourcify";
 
 class Vyper implements IVyperCompiler {
+  constructor(
+    private vyperRepoPath: string,
+  ) {}
+
   async compile(
     version: string,
     vyperJsonInput: VyperJsonInput
   ): Promise<VyperOutput> {
-    return await vyperCompiler.compile(version, vyperJsonInput);
+    return await useVyperCompiler(
+      this.vyperRepoPath,
+      version,
+      vyperJsonInput,
+    );
   }
 }
-const vyper = new Vyper()
+const vyper = new Vyper(
+  "/path/to/vyper",
+);
+```
 
-// Pass the class to create a VyperCheckedContract
-const checkedContract = new VyperCheckedContract(
-  vyper,
-  "0.3.7", // vyper version
-  "contract.vy", // contract path
-  "MyContract", // contract name
-  {/** vyper settings */},
-  {/** vyper source files */}
-)
+## SourcifyChain
+
+The `SourcifyChain` class is a crucial component that provides a standardized and resilient way to interact with blockchain networks during smart contract verification. It handles multiple aspects of blockchain interaction, in particular:
+
+### Core Features
+- **Resilient RPC management**
+  - Manages multiple RPC providers for a single chain with automatic fallback
+  - Implements sequential provider retry logic to ensure reliable network access
+  - Gracefully handles provider failures without disrupting verification flow
+
+- **Comprehensive bytecode retrieval**
+  - Fetches both runtime and creation bytecode
+  - Supports advanced factory contract scenarios through trace API integration
+
+- **Robust error handling**
+  - Implements configurable timeout handling to prevent hanging on unresponsive RPCs
+  - Provides detailed error reporting with provider-specific context
+  - Maintains connection stability through automatic retry mechanisms
+
+### Creating a SourcifyChain Instance
+
+```typescript
+import { SourcifyChain } from "@ethereum-sourcify/lib-sourcify";
+
+// Create a SourcifyChain for Ethereum Mainnet
+const mainnetChain = new SourcifyChain({
+  name: "Ethereum Mainnet",
+  chainId: 1,
+  rpc: ["https://eth.llamarpc.com"],
+  supported: true
+});
+
+// Create a SourcifyChain for a local Hardhat network
+const hardhatChain = new SourcifyChain({
+  name: "Hardhat Network Localhost",
+  shortName: "Hardhat Network",
+  chainId: 31337,
+  rpc: ["http://localhost:8545"],
+  supported: true
+});
+```
+
+### SourcifyChain Properties
+
+- `name`: The human-readable name of the chain
+- `chainId`: The numeric chain identifier
+- `rpc`: An array of RPC endpoints (strings or FetchRequest objects)
+- `supported`: Whether the chain is supported for verification
+- `providers`: JsonRPC providers initialized from the RPC endpoints
```

Comment:
I don't think they need to know this. This field is not passed into the constructor
```suggestion
```